plugins/cpu/fu-plugin-cpu.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2020 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+#include "fu-cpu-device.h"
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_coldplug (FuPlugin *plugin, GError **error)
+{
+	gsize length;
+	g_autofree gchar *data = NULL;
+	g_auto(GStrv) lines = NULL;
+
+	if (!g_file_get_contents ("/proc/cpuinfo", &data, &length, error))
+		return FALSE;
+
+	lines = g_strsplit (data, "\n\n", 0);
+	for (guint i = 0; lines[i] != NULL; i++) {
+		g_autoptr(FuCpuDevice) dev = NULL;
+		g_strstrip(lines[i]);
```

Comment:
is this needed?

Reply:
Yeah; I was seeing an extra device being created with no data.  This was my way to avoid it.  might be a cleaner way though.

Reply:
Not needed, will remove.